scipy/optimize/tests/_cython_examples/meson.build

```diff
@@ -0,0 +1,18 @@
+project('random-build-examples', 'c', 'cpp', 'cython')
+
+py_mod = import('python')
+py3 = py_mod.find_installation(pure: false)
+
+cc = meson.get_compiler('c')
+cy = meson.get_compiler('cython')
+
+# Keep synced with pyproject.toml
+if not cy.version().version_compare('>=3.0.8')
+  error('tests requires Cython >= 3.0.8')
```

Comment:
minor: inconsistent indentation between this line and `py3.extension_module` below. Using 4 spaces everywhere is fine, as is 2 (I have no strong preference) - as long as it's consistent.

This comment applies to the other `meson.build` files too.